test/e2e/subscription_e2e_test.go

```diff
@@ -1218,6 +1197,18 @@ var _ = Describe("Subscription", func() {
 				require.FailNowf(GinkgoT(), "failed", "subscription contains unexpected installplan condition: %v", cond)
 			case operatorsv1alpha1.SubscriptionInstallPlanMissing:
 				require.Equal(GinkgoT(), operatorsv1alpha1.ReferencedInstallPlanNotFound, cond.Reason)
+			case operatorsv1alpha1.SubscriptionCatalogSourcesUnhealthy:
```

Comment:
The way this test is written makes it sensitive to _any_ change in the cluster state or the reconciliation flow of the controller. Since we're trying to test a negative ("all of the non-`InstallPlan` conditions can change between the time we looked at them in the beginning of the test and now"), I don't know that this test is super powerful, and, since that kind of approach is very fragile, I wonder if a different approach exists (maybe just to delete the test? what is it testing?).

Consider the case where a CatalogSource happens to blip unhealthy for a moment during this check. Why should the test fail if the reconciler has correctly exposed that downtime in status?